ec-service/src/main/java/org/dcsa/endorsementchain/components/eblenvelope/EblEnvelopeList.java

```diff
@@ -0,0 +1,43 @@
+package org.dcsa.endorsementchain.components.eblenvelope;
+
+import lombok.experimental.UtilityClass;
+import org.dcsa.endorsementchain.persistence.entity.EblEnvelope;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+@UtilityClass
+public class EblEnvelopeList {
+  public List<EblEnvelope> order(List<EblEnvelope> envelopeList) {
+    EblEnvelope firstEnvelope = first(envelopeList).orElse(null);
+
+    List<EblEnvelope> sortedList = new ArrayList<>();
+    EblEnvelope envelope = firstEnvelope;
+    while (envelope != null) {
+      sortedList.add(envelope);
+      envelope = next(envelopeList, envelope.getEnvelopeHash()).orElse(null);
+    }
```

Comment:
This is `O(n^2)` (because `next` is `O(n)`).  Since performance is not critical, I think it is fine to leave this as a TODO comment for now and move on.

Reply:
fixed